src/app/issueForm/issueForm.controller.ts

```diff
@@ -18,45 +18,83 @@ module testApp {
   export class IssueFormController {
     public awesomeThings: Thing[];
     public search: string;
-    private pendingTask: any;
-    private $http: any;
-    private $window: any;
-    private $stateParams: any;
+    public mode;
+    public issueNumber: number;
     public repository;
     public owner;
     public repo;
+    public issue: any;
+
+    private $http: any;
+    private HttpService: any;
+    private $window: any;
+    private $stateParams: any;
+    private $state: any;
 
     submitForm(isValid, issue) {
       if(!isValid) {
         alert("Not valid");
       } else {
         var authToken;
-        var _this = this;
         var params = {};
         params['title'] = issue.title;
         params['body'] = issue.body;
-        this.$http.post(
+        var promise = null;
+
+        if(this.mode){
+          promise = this.HttpService.call('PATCH', "repos/" + this.owner + "/" + this.repo +  "/issues/" + this.issueNumber, params)
+          promise = this.$http.patch(
```

Comment:
yes, it was working only because it was doing the same thing twice. I've changed it for the next push